src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonNode/JsonArrayTests.cs

```diff
@@ -486,5 +487,210 @@ public static void LazyInitializationIsThreadSafe()
+                {
+                    JsonArray array2 = JsonArray.Create(document2.RootElement);
+                    Assert.True(JsonNode.DeepEquals(array, array2));
+                }
+                using (JsonDocument document3 = JsonDocument.Parse("[2, 1, 4]"))
+                {
+                    JsonArray array2 = JsonArray.Create(document3.RootElement);
+                    Assert.False(JsonNode.DeepEquals(array, array2));
+                }
+            }
+        }
+
+        [Fact]
+        public static void UpdateClonedObjectNotAffectOriginal()
+        {
+            var jArray = new JsonArray(10, 20);
+            
+            var clone = jArray.DeepClone().AsArray();
+            clone[1] = 3;
+
+            Assert.Equal(20, jArray[1].GetValue<int>());
+        }
+
+        [Fact]
+        public static void GetValueKind()
+        {
+            Assert.Equal(JsonValueKind.Array, new JsonArray().GetValueKind());
+        }
+
+        [Fact]
+        public static void GetElementIndex()
+        {
+            var trueValue = JsonValue.Create(true);
+            var falseValue = JsonValue.Create(false);
+            var numberValue = JsonValue.Create(15);
+            var stringValue = JsonValue.Create("ssss");
+            var nestedObject = new JsonObject();
+            var nestedArray = new JsonArray();
+
+            var array = new JsonArray();
+            array.Add(trueValue);
+            array.Add(falseValue);
+            array.Add(numberValue);
+            array.Add(stringValue);
+            array.Add(nestedObject);
+            array.Add(nestedArray);
+
+            Assert.Equal(0, trueValue.GetElementIndex());
+            Assert.Equal(1, falseValue.GetElementIndex());
+            Assert.Equal(2, numberValue.GetElementIndex());
+            Assert.Equal(3, stringValue.GetElementIndex());
+            Assert.Equal(4, nestedObject.GetElementIndex());
+            Assert.Equal(5, nestedArray.GetElementIndex());
+        }
+
+        [Fact]
+        public static void GetValues_ValueType()
+        {
+            JsonArray jsonArray = new JsonArray(1, 2, 3, 2);
+            
+            IEnumerable<int> values = jsonArray.GetValues<int>();
+
+            Assert.Equal(jsonArray.Count, values.Count());
+            Assert.Equal(1, values.ElementAt(0));
+            Assert.Equal(2, values.ElementAt(1));
+            Assert.Equal(3, values.ElementAt(2));
+            Assert.Equal(2, values.ElementAt(3));
+
+            jsonArray = new JsonArray(1, null);
+            Assert.Throws<NullReferenceException>(() => jsonArray.GetValues<int>().Count());
+        }
+
+        [Fact]
+        public static void GetValues_ReferenceType()
+        {
+            var student1 = new Student();
+            var student2 = new Student();
+            JsonArray jsonArray = new JsonArray(JsonValue.Create(student1), null, JsonValue.Create(student2));
```

Comment:
Could you also add a test enumerating `null` for value types and asserting the right exception is being thrown?

Reply:
Nevermind, I just realized you already do that in line 660.